src/Blocks/BlockCauldron.h

```diff
@@ -86,29 +94,45 @@ class cBlockCauldronHandler :
 		return true;
 	}
 
+
+
+
+
 	virtual bool IsUseable() override
 	{
 		return true;
 	}
 
-	virtual void OnUpdate(cChunkInterface & a_ChunkInterface, cWorldInterface & a_WorldInterface, cBlockPluginInterface & a_PluginInterface, cChunk & a_Chunk, int a_RelX, int a_RelY, int a_RelZ) override
+
+
+
+
+	virtual void OnUpdate(
+		cChunkInterface & a_ChunkInterface,
+		cWorldInterface & a_WorldInterface,
+		cBlockPluginInterface & a_PluginInterface,
+		cChunk & a_Chunk,
+		const Vector3i a_RelPos
+	) override
 	{
-		int BlockX = a_RelX + a_Chunk.GetPosX() * cChunkDef::Width;
-		int BlockZ = a_RelZ + a_Chunk.GetPosZ() * cChunkDef::Width;
-		if (!a_WorldInterface.IsWeatherWetAt(BlockX, BlockZ) || (a_RelY != a_WorldInterface.GetHeight(BlockX, BlockZ)))
+		auto WorldPos = a_Chunk.RelativeToAbsolute(a_RelPos);
+		if (!a_WorldInterface.IsWeatherWetAtXYZ(WorldPos))
```

Comment:
I think `IsWeatherWetAtXYZ` classes it as dry if there is a block at the exact coordinate queried.
```suggestion
		if (!a_WorldInterface.IsWeatherWetAtXYZ(WorldPos.addedY(1)))
```